packages/amplify-graphql-model-transformer/src/resources/dynamo-model-resource-generator.ts

```diff
@@ -0,0 +1,301 @@
+import * as cdk from '@aws-cdk/core';
+import { TransformerContextProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import {
+  ModelResourceIDs,
+  ResourceConstants,
+  SyncResourceIDs,
+} from 'graphql-transformer-common';
+import { ObjectTypeDefinitionNode } from 'graphql';
+import { SyncUtils, TransformerNestedStack } from '@aws-amplify/graphql-transformer-core';
+import {
+  AttributeType,
+  CfnTable,
+  StreamViewType,
+  Table,
+  TableEncryption,
+} from '@aws-cdk/aws-dynamodb';
+import * as iam from '@aws-cdk/aws-iam';
+import { CfnDataSource } from '@aws-cdk/aws-appsync';
+import { CfnRole } from '@aws-cdk/aws-iam';
+import { ModelResourceGenerator } from './model-resource-generator';
+import { DynamoDBModelVTLGenerator, ModelVTLGenerator } from '../resolvers';
+
+/**
+ * DynamoModelResourceGenerator is an implementation of ModelResourceGenerator,
+ * providing necessary utilities to generate the DynamoDB resources for models
+ */
+export class DynamoModelResourceGenerator extends ModelResourceGenerator {
+  protected generatorType = 'DynamoModelResourceGenerator';
+
+  generateResources(ctx: TransformerContextProvider): void {
+    if (!this.isEnabled()) {
+      return;
+    }
+
+    if (this.isProvisioned()) {
```

Comment:
I think we're making an assumption here that DDB resources are already provisioned. I would say a better check would be to check the dataSource instead.